src/Microsoft.IdentityModel.Xml/EnvelopedSignatureWriter.cs

```diff
@@ -179,7 +179,7 @@ private Signature CreateSignature()
             var hashAlgorithm = cryptoProviderFactory.CreateHashAlgorithm(digestAlgorithm);
 
             if (hashAlgorithm == null)
-                throw LogExceptionMessage(new XmlValidationException(FormatInvariant(LogMessages.IDX30213, cryptoProviderFactory.ToString(), _signingCredentials.Digest)));
+                throw LogExceptionMessage(new XmlValidationException(FormatInvariant(LogMessages.IDX30213, MarkAsNonPII(cryptoProviderFactory.ToString()), _signingCredentials.Digest)));
```

Comment:
Doesn't looke like CryptoProviderFactory exposes a ToString override, does this just show the type? Is it better to just make that explicit here?

Reply:
CryptoProviderFactory is not sealed, and application-provided SigningCredentials.CryptoProviderFactory could override ToString() and possibly return PII from there. The IDX30213 message contains `The CryptoProviderFactory: '{0}'`.

Perhaps it's best to use just `cryptoProviderFactory` as the log argument, with neither ToString nor MarkAsNonPII. If ShowPII == false, then the name of the type will be logged anyway.

Reply:
Updated to display the class name of cryptoProviderFactory